docs/source/ContributionGuide.rst

```diff
@@ -65,10 +65,119 @@ be submitted as a github Pull Request. This process tries to ensure that every
 contributed patch goes through the CI jobs before it is considered good for
 inclusion.
 
+Git workflow
+------------
+
+- Fork the repository in github.
+
+- Clone from your fork::
+
+    $ git clone git@github.com:<username>/avocado.git
+
+- Enter the directory::
+
+    $ cd avocado
+
+- Create a ``remote``, pointing to the upstream::
+
+    $ git remote add upstream git@github.com:avocado-framework/avocado-misc-tests.git
+
+- Configure your name and e-mail in git::
+
+    $ git config --global user.name "Your Name"
+    $ git config --global user.email email@foo.bar
+
+- Golden tip: never work on local branch master. Instead, create a new
+  local branch and checkout to it::
+
+    $ git checkout -b my_new_local_branch
+
+- Code and then commit your changes::
+
+    $ git add new-file.py
+    $ git commit -a -s
+
+- Write a good commit message, pointing motivation, issues that you're
+  addressing. Usually you should try to explain 3 points in the commit
+  message: motivation, approach and effects::
+
+    header          <- Limited to 50 characters. No period.
+                    <- Blank line
+    message         <- Any number of lines, limited to 72 characters per line.
+                    <- Blank line
+    Reference:      <- External references, one per line (issue, trello, ...)
+    Signed-off-by:  <- Signature (created by git commit -s)
+
+- Make sure your code is working (install your version of avocado, test
+  your change, run ``make check`` to make sure you didn't introduce any
+  regressions).
+
+- Paste the ``job.log`` file content from the previous step in a pastebin
+  service, like fpaste.org.
```

Comment:
One little tip that may be useful to include (in a "hint" like section) is to install/use the `fpaste` utility.  Running `fpaste ~/avocado/job-results/latest/job.log` can be rather convenient.